cmd/devdata/system_intake.go

```diff
@@ -0,0 +1,404 @@
+package main
+
+import (
+	"context"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/guregu/null"
+	"go.uber.org/zap"
+
+	"github.com/cmsgov/easi-app/cmd/devdata/mock"
+	"github.com/cmsgov/easi-app/pkg/graph/model"
+	"github.com/cmsgov/easi-app/pkg/graph/resolvers"
+	"github.com/cmsgov/easi-app/pkg/models"
+	"github.com/cmsgov/easi-app/pkg/storage"
+)
+
+// creates, fills out the initial request form, and submits a system intake
+func makeSystemIntakeAndSubmit(
+	requestName string,
+	intakeID *uuid.UUID,
+	requesterEUAID string,
+	logger *zap.Logger,
+	store *storage.Store,
+) *models.SystemIntake {
+	intake := makeSystemIntake(requestName, intakeID, requesterEUAID, logger, store)
+	return submitSystemIntake(logger, store, intake)
+}
+
+// creates an intake and fills out the initial request form
+func makeSystemIntake(
+	requestName string,
+	intakeID *uuid.UUID,
+	requesterEUAID string,
+	logger *zap.Logger,
+	store *storage.Store,
+) *models.SystemIntake {
+	intake := createSystemIntake(
+		intakeID,
+		logger,
+		store,
+		requesterEUAID,
+		"Requester Name",
+		models.SystemIntakeRequestTypeNEW,
+	)
+	return fillOutInitialIntake(requestName, logger, store, intake)
+}
+
+// updates an intake and fills out the initial request form
+func fillOutInitialIntake(
+	requestName string,
+	logger *zap.Logger,
+	store *storage.Store,
+	intake *models.SystemIntake,
+) *models.SystemIntake {
+	intake = updateSystemIntakeRequestDetails(logger, store, intake,
+		requestName,
+		"An intense business need",
+		"with a great business solution",
+		true,
+		"Some CEDAR System ID",
+		"the current stage",
+		true,
+	)
+	updateSystemIntakeContact(logger, store, intake,
+		"USR1",
+		"Center for Medicare",
+		"Requester",
+	)
+	createSystemIntakeContact(logger, store, intake,
+		"A11Y",
+		"Center for Medicare",
+		"Business Owner",
+	)
+	createSystemIntakeContact(logger, store, intake,
+		"OQYV",
+		"Center for Medicare",
+		"Product Manager",
+	)
+	createSystemIntakeContact(logger, store, intake,
+		"GP87",
+		"Center for Medicare",
+		"ISSO",
+	)
+	intake = updateSystemIntakeContactDetails(logger, store, intake,
+		"User One",
+		"Office of the Actuary",
+		"Ally Anderson",
+		"Office of Minority Health",
+		"Hallie O'Hara",
+		"Emergency Preparedness and Response Operations",
+		true,
+		"Leatha Gorczany",
+	)
+	return updateSystemIntakeContractDetails(logger, store, intake)
+}
+
+func submitSystemIntake(
+	logger *zap.Logger,
+	store *storage.Store,
+	intake *models.SystemIntake,
+) *models.SystemIntake {
+	ctx := mock.CtxWithLoggerAndPrincipal(logger, intake.EUAUserID.ValueOrZero())
+	// until the submit function is refactored out of services, manually submit
+	// NOTE: does not send emails
+	mockSubmitIntake := func(ctx context.Context, intake *models.SystemIntake, action *models.Action) error {
+		updatedTime := time.Now()
+		intake.SubmittedAt = &updatedTime
+		intake.UpdatedAt = &updatedTime
+
+		// TODO: Remove when Admin Actions v2 is live
```

Comment:
Ah, this is a side effect of the way System Intakes and Business Cases are written in services for purposes of the older REST endpoints. Currently the resolver must be given the services function to submit an intake which does various things. Instead of pulling that package in, I decided to just copy over the core functionality.

I think once we refactor the REST endpoints to become GQL resolvers AND not rely on the older services code that this can be updated to just call the resolver.